src/server/game/DataStores/DBCEnums.h

```diff
@@ -381,41 +410,56 @@ enum CriteriaTypes
     CRITERIA_TYPE_LEVEL_BATTLE_PET                      = 160,
     CRITERIA_TYPE_CAPTURE_BATTLE_PET_CREDIT             = 161, // triggers a quest credit
     CRITERIA_TYPE_LEVEL_BATTLE_PET_CREDIT               = 162, // triggers a quest credit
     CRITERIA_TYPE_ENTER_AREA                            = 163, // triggers a quest credit
     CRITERIA_TYPE_LEAVE_AREA                            = 164, // triggers a quest credit
     CRITERIA_TYPE_COMPLETE_DUNGEON_ENCOUNTER            = 165,
+    // 166 - unused (Legion - 23420)
     CRITERIA_TYPE_PLACE_GARRISON_BUILDING               = 167,
     CRITERIA_TYPE_UPGRADE_GARRISON_BUILDING             = 168,
     CRITERIA_TYPE_CONSTRUCT_GARRISON_BUILDING           = 169,
     CRITERIA_TYPE_UPGRADE_GARRISON                      = 170,
     CRITERIA_TYPE_START_GARRISON_MISSION                = 171,
-    // 172
+    CRITERIA_TYPE_START_ORDER_HALL_MISSION              = 172,
     CRITERIA_TYPE_COMPLETE_GARRISON_MISSION_COUNT       = 173,
     CRITERIA_TYPE_COMPLETE_GARRISON_MISSION             = 174,
     CRITERIA_TYPE_RECRUIT_GARRISON_FOLLOWER_COUNT       = 175,
-    // 176
-    // 177
+    CRITERIA_TYPE_RECRUIT_GARRISON_FOLLOWER             = 176,
+    // 177 - 0 criterias (Legion - 23420)
     CRITERIA_TYPE_LEARN_GARRISON_BLUEPRINT_COUNT        = 178,
-    // 179
-    // 180
-    // 181
+    // 179 - 0 criterias (Legion - 23420)
+    // 180 - 0 criterias (Legion - 23420)
+    // 181 - 0 criterias (Legion - 23420)
     CRITERIA_TYPE_COMPLETE_GARRISON_SHIPMENT            = 182,
     CRITERIA_TYPE_RAISE_GARRISON_FOLLOWER_ITEM_LEVEL    = 183,
     CRITERIA_TYPE_RAISE_GARRISON_FOLLOWER_LEVEL         = 184,
     CRITERIA_TYPE_OWN_TOY                               = 185,
     CRITERIA_TYPE_OWN_TOY_COUNT                         = 186,
-    CRITERIA_TYPE_RECRUIT_GARRISON_FOLLOWER             = 187,
-    CRITERIA_TYPE_OWN_HEIRLOOMS                         = 189
+    CRITERIA_TYPE_RECRUIT_GARRISON_FOLLOWER_WITH_QUALITY= 187,
+    // 188 - 0 criterias (Legion - 23420)
+    CRITERIA_TYPE_OWN_HEIRLOOMS                         = 189,
+    CRITERIA_TYPE_ARTIFACT_POWER_EARNED                 = 190,
+    CRITERIA_TYPE_ARTIFACT_TRAITS_UNLOCKED              = 191,
+    CRITERIA_TYPE_HONOR_LEVEL_REACHED                   = 194,
+    CRITERIA_TYPE_PRESTIGE_REACHED                      = 195,
+    // 196 - CRITERIA_TYPE_REACH_LEVEL_2 or something
+    // 197 - Order Hall Advancement related
+    CRITERIA_TYPE_ORDER_HALL_TALENT_LEARNED             = 198,
+    CRITERIA_TYPE_APPEARANCE_UNLOCKED_BY_SLOT           = 199,
+    CRITERIA_TYPE_ORDER_HALL_RECRUIT_TROOP              = 200,
+    // 201 - 0 criterias (Legion - 23420)
+    // 202 - 0 criterias (Legion - 23420)
+    CRITERIA_TYPE_COMPLETE_WORLD_QUEST                  = 203,
+    // 204 - Special criteria type to award players for some external events? Comes with what looks like an identifier, so guessing it's not unique.
 };
 
 #define CRITERIA_TYPE_TOTAL 190
```

Comment:
poke
port this assert
https://github.com/TrinityCore/TrinityCore/blob/3.3.5/src/server/game/Achievements/AchievementMgr.cpp#L2270